students/thomasvad/lesson4/mailroom2.py

```diff
@@ -0,0 +1,106 @@
+import sys
+import os
+
+donors = {
+'William Gizzard': [1.00, 1234.12, 8000],
+'Mark Zhophone': [120000.00],
+'Jeff Binsue': [80.00, 200.00, 50.00],
+'Prince luis': [666],
+'Paul Alad': [56743, 300000, 10000000, 88743],
+'mike mixer': [309000, 10000000]}
+
+
+
+def thank():
+    name = input("Please enter the donors full name. To check the list of previous donors type list: ")
+    print('\n')
+    while name.lower() == 'list':
+        print('Past Donors:')
+        for donor in donors.keys():
+            print(donor)
+        print('\n')
+        name = input('Please enter the full name of the donor. If it is a new donor enter their name: ')
+        print('\n')
+
+    # compares each donor name with what was inputed as a set to get the correct key if donor is in list
+    decide = sorted([(i, set(i.lower()) == set(name.lower())) for i in donors.keys()], key= lambda x: x[1])
+    decide2 =  [bool[1] for bool in decide]
+#    print(decide, decide2)
```

Comment:
to check if the name is in the dictionary just use `.get()` function.

`if donors.get(name)`

if you're worried about case sensitive strings you can store them all lower case and then use `capitalize()` on the string for display.